FEM/pcs_dm.cpp

```diff
@@ -828,7 +828,9 @@ double CRFProcessDeformation::Execute(int loop_process_number)
 #if defined(USE_MPI) || defined(USE_PETSC)
 				}
 #endif
-				if (Error <= Tolerance_global_Newton)//testing only relative residual so far
+				if (Error <= Tolerance_global_Newton && Norm <= m_num->nls_abs_residual_tolerance
```

Comment:
All norms are set to num_limits():max initially, so only the Tolerance_global_Newton is active in the standard case (status of master branch). The user in effect decides which tolerance to use by setting it to a small value, while keeping the others high. Only if you set all four to small values you will end up with the strict case that you mentioned.